app/releases/Releases.tsx

```diff
@@ -0,0 +1,168 @@
+import { ReleasesResponse } from "@/lib/types";
+import { Repository } from "@/lib/types";
+import { Release } from "@/lib/types";
+import Link from "next/link";
+import Markdown from "@/components/Markdown";
+import { FiExternalLink, FiGithub } from "react-icons/fi";
+import LoadingText from "@/components/LoadingText";
+
+export default async function Releases(props: { className?: string }) {
+  const accessToken = process.env.GITHUB_PAT;
+
+  if (!accessToken) {
+    if (process.env.NODE_ENV === "development") {
+      console.error("'GITHUB_PAT' is not configured in the environment.");
+      return (
+        <>
+          <span className="flex w-full justify-center text-gray-600 dark:text-gray-400 text-lg font-semibold py-10">
+            No recent releases
+          </span>
+        </>
+      );
+    }
+
+    throw "'GITHUB_PAT' is not configured in the environment.";
+  }
+
+  const response = await fetch("https://api.github.com/graphql", {
+    method: "POST",
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      query: `{
+          organization(login: "${process.env.NEXT_PUBLIC_GITHUB_ORG}") {
+            repositories(first: 100) {
+              nodes {
+                name
+                releases(first: 10, orderBy: {field: CREATED_AT, direction: DESC}) {
+                  nodes {
+                    name
+                    createdAt
+                    description
+                    url
+                    author {
+                      login
+                      avatarUrl
+                    }
+                    mentions (first: 10) {
+                      nodes {
+                        login 
+                        avatarUrl
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }`,
+    }),
+  });
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+
+  const json = (await response.json()) as ReleasesResponse;
+  const data = json.data;
+
+  const repositories: Repository[] = data.organization.repositories.nodes;
+  const allReleases: Release[] = [];
+  for (const repository of repositories) {
+    for (const release of repository.releases.nodes) {
+      release.repository = repository.name;
+      allReleases.push(release);
+    }
+  }
+
+  const sortedReleases = allReleases.sort(
+    (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+  );
```

Comment:
Move this release fetching logic to a shared function that both the home page releases and `/releases` page can re-use instead of redefining it twice. Pass the limit as an arg.

Reply:
made the changes